wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -277,7 +309,11 @@ public Void call() throws IOException {
         }
 
         TypeSpec typeSpec = javaGenerator.generateType(type);
-        ClassName javaTypeName = (ClassName) javaGenerator.typeName(type.type());
+        TypeName typeName = javaGenerator.protoFieldsTypeName(type.type());
```

Comment:
Hang on, why is this still necessary? I was expecting javaGenerator.typeName() to always do the right thing.